### PR TITLE
IMPROV: Strassesn's multiplication method

### LINKED ISSUE #1181: optimize matrix operations using Strassen algorithm ?
Adding this one here, as it's quite interesting :

Strassen multiplication algorithm is faster, with a complexity of approximately O(n^2.8074) , compared to O(n^3) for standard multiplication algorithm.

Strassen  Matrix multiplication is based on a divide and conquer-based approach, by diving matrix into smaller square matrix 
[https://www.topcoder.com/thrive/articles/strassenss-algorithm-for-matrix-multiplication](https://www.topcoder.com/thrive/articles/strassenss-algorithm-for-matrix-multiplication)

### PR DESCRIPTION
**Pull request by sudarshanmg**
- Added `fastMul()` method that uses Strassen's algorithm
- Trying to close #1181

### CODE REVIEW

**Comment by obiot** [comment]
that was fast ! I'll try tomorrow :)

Looking at the implementation it is longer than a regular multiplication though, makes me wonder if it actually faster, I'll let you know tomorrow anyway, I'll do some benchmark !

File: src/math/matrix3.js [hunk 1]
```diff
@@ -197,6 +197,82 @@ export default class Matrix3d {
...
+   * multiply both matrix
+   * @param {Matrix3d} m - Other matrix
+   * @returns {Matrix3d} Reference to this object for method chaining
+   */
+
+  fastMul(m) {
+    let a = this.val;
+    let b = m.val;
+
+    const A11 = [a[0], a[1], a[4], a[5]];
```
Comment (obiot) [inline src/math/matrix3.js:241]: Is there a way to do the same without creating temporary arrays?

**Comment by sudarshanmg** [comment]
### **You're right!** its taking the same amount of time. (sometimes even longer)
I will try to figure out a way to optimize it. Let me know if you any ideas (This is new for me). :smiling_face_with_tear:

**Comment by obiot** [comment]
@sudarshanmg did you find the time to look at how to optimise it ? if not I will close this PR for now. Feel free to reopen one if you find a way to do it ? thanks again !

**Pull request closed without merging**